pkg/skaffold/jib/jib.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jib
+
+import (
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/pkg/errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// GetDependenciesMaven finds the source dependencies for the given jib-maven artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesMaven(workspace string, _ /*a*/ *v1alpha3.JibMavenArtifact, isWindows bool) ([]string, error) {
+	return getDependencies(workspace, "pom.xml", "mvn", "mvnw.cmd", isWindows, []string{"jib:_skaffold-files", "-q"}, "jib-maven")
+}
+
+// GetDependenciesGradle finds the source dependencies for the given jib-gradle artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesGradle(workspace string, _ /*a*/ *v1alpha3.JibGradleArtifact, isWindows bool) ([]string, error) {
+	return getDependencies(workspace, "build.gradle", "gradle", "gradlew.bat", isWindows, []string{"_jibSkaffoldFiles", "-q"}, "jib-gradle")
+}
+
+func exists(workspace string, filename string) bool {
+	_, err := os.Stat(filepath.Join(workspace, filename))
+	return err == nil
+}
+
+func getDependencies(workspace string, buildFile string, defaultExecutable string, windowsExecutable string, isWindows bool, subCommand []string, artifactName string) ([]string, error) {
+	if !exists(workspace, buildFile) {
+		return nil, errors.Errorf("no %s found", buildFile)
+	}
+
+	executable := defaultExecutable
+	if isWindows {
+		if exists(workspace, windowsExecutable) {
+			executable = "call"
+			subCommand = append([]string{windowsExecutable}, subCommand...)
```

Comment:
Do we need to use `call` from within Go?

Reply:
Ah, I actually need to use `cmd.exe /C ...` instead. Also, @GoogleContainerTools/container-tools is there a style preference for using back-ticks for string constants vs quotes?

Reply:
Not that I'm aware of: we generally follow standard Go & Kubernetes Coding Conventions: https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md

In practice, raw string literals (back-ticks) in Go are only used for strings containing a back-slash or double quotes. Other string declarations should probably use the interpreted string literal form (double quotes) unless there is a special requirement to do so.

Reply:
Thanks for clarifying!